src/cpp/rtps/transport/TCPAcceptor.cpp

```diff
@@ -32,6 +32,7 @@ TCPAcceptor::TCPAcceptor(
     , io_service_(&io_service)
 {
     endpoint_ = asio::ip::tcp::endpoint(parent->generate_protocol(), IPLocator::getPhysicalPort(locator_));
+    locator_.port = acceptor_.local_endpoint().port();
```

Comment:
I think you should probably add this line also in the other TCPAcceptor Constructor (the following one at line 38 in which one of the arguments is 'interface' instead of 'parent'), because in the function `TCPTransportInterface::create_acceptor_socket()` it is considered also the case in which the whitelist is not empty and the TCPAcceptor is defined through the other constructor.
Maybe the lines 34 and 35 should be switched because the object locator_  is used in `IPLocator::getPhysicalPort()` and then after that its port is updated.